### PR TITLE
results page now working with GET API, tests and error handling included

### LINKED ISSUE #64: Survey Results Page
- Admin/Instructor View
    - Get list of associated TAs, can filter by term and course.
    - Select TA to go into that TA's view
    - TA View (Starting view if logged in as a TA)
        - Get list of survey instances
        - View responses of an individual survey

### PR DESCRIPTION
**Pull request by F-Ting**
fixes #64 

Added Vue-Test-Utils for testing,
Completed basic components for Results page,
Added routing to main.js

### CODE REVIEW

File: src/views/components/response_components/response_page.vue [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+            survey_instance_id,
+            user_id,
+            selected_ta_id,
+            term,
+            course
+        ) {
+            this.loading = true;
+            this.no_response = false;
+            let url =
+                "http://localhost:3000/get_info.php?what=survey_results&user_id=" +
```
Comment (siefkenj) [inline src/views/components/response_components/response_page.vue]: Don't prefix with `http://localhost:3000/`

File: src/views/components/response_components/select_ta.vue [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        // return false if object is null or empty
+        isEmpty: function(object) {
+            return !object || Object.keys(object).length === 0;
+        }
+    },
+    computed: {
+        //returns the updated component of term and course combined
+        // added date to be unique every time
+        term_and_course() {
+            return this.term, this.course, Date.now();
```
Comment (siefkenj) [inline src/views/components/response_components/select_ta.vue]: Don't use the comma operator. Are you trying to return an array?

Comment (F-Ting) [reply src/views/components/response_components/select_ta.vue]: the computed property changes when either term or course is changing(from the select element), this just returns a unique value. Then the watch property sees a change in the computed value and updates the page based on new specifications of term and course.

Comment (F-Ting) [reply src/views/components/response_components/select_ta.vue]: I can change it to an object containing term and course

Comment (siefkenj) [reply src/views/components/response_components/select_ta.vue]: IF you would like to call `this.term` and `this.course`, do

```
this.term;
this.course;
return Date.now();
```

File: src/views/components/response_components/select_ta.vue [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+        isEmpty: function(object) {
+            return !object || Object.keys(object).length === 0;
+        }
+    },
+    computed: {
+        //returns the updated component of term and course combined
+        // added date to be unique every time
+        term_and_course() {
+            this.term;
+            this.course;
```
Comment (siefkenj) [inline src/views/components/response_components/select_ta.vue]: I still don't believe you. Are you claiming that `this.term` is a dynamically computed property? Those are very uncommon in JS...

Comment (F-Ting) [reply src/views/components/response_components/select_ta.vue]: `this.term` isn't a dynamically computed property, `term_and_course` is. Vue creates this dependency relationship such that it knows `term_and_course` depends on `this.term` and `this.course`, and thus any changes to it's dependencies will trigger the function. As the Vue documentation states "A computed property will only re-evaluate when some of its dependencies have changed.".

Comment (siefkenj) [reply src/views/components/response_components/select_ta.vue]: Then removing these lines shouldn't make a difference. Can you confirm this?

Comment (F-Ting) [reply src/views/components/response_components/select_ta.vue]: The entire function? Also dependencies must to be reactive, which means that Date.now() by itself will never trigger an update. I tried with different combinations within this function, this is the one works best if you didn't want comma separated return statements.

Comment (siefkenj) [reply src/views/components/response_components/select_ta.vue]: This cannot be right. Figure out what is actually going on and what is actually needed. As it is, you have two NOOP  (https://en.wikipedia.org/wiki/NOP) instructions followed by a return statement. The NOOPs shouldn't be there.

Comment (F-Ting) [reply src/views/components/response_components/select_ta.vue]: After a little bit of digging into Vue's source code and community, here's what I got out of it: 
1. All `data` properties are defined as getters and setters in Vue, which are also considered `reactive' properties of the particular Vue instance.
2. On declaration of a computed property, Vue associates all 'reactive' getters in the computed property into the dependency list of this computed property (supported by the `Dep` dependency tracker in Vue source code). Which allows the computed property to watch the list for change.
3. All properties requesting the computed property will also add the computed property to it's dependency list. 
4. Despite the variables are noop, they are just used associated with the dependency list on initialization of the computed property.
5. Vuejs creator commented on another way to do this without having to use the computed property *cries*. but I find it as an easy way to collectively watch a set of variables we use for filtering.

Comment (siefkenj) [reply src/views/components/response_components/select_ta.vue]: Okay. Add a comment saying that these are accessed via getters and therefore have sideeffects in Vue and that this is on purpose with a link to the discussion you read.

Comment (F-Ting) [reply src/views/components/response_components/select_ta.vue]: If this makes no sense, I will switch to only using the `watch` property.

**Pull request merged**